### DISCUSSION TITLE
由于按下键盘Shift + 任意按键报错引发的提问

### DISCUSSION THREAD

**Discussion by hou-pu** [alibaba/designable#200] [Q&A] [answered]
想请教个问题，我在designable项目运行页面中
按下shift + 任意按钮会报错：
![download](https://user-images.githubusercontent.com/19144982/149774794-792f234f-ded7-4558-8798-3124696773f2.png)

我看到源码里是有这样一段逻辑：
packages/core/src/models/Shortcut.ts:37
![download](https://user-images.githubusercontent.com/19144982/149774849-e05588ff-adf6-4adf-908f-34ec02545a29.png)

然后我找到了下面这个配置，因为只配置了一个codes，所以导致上面的错误：
packages/core/src/shortcuts/MultiSelection.ts:8
![download-1](https://user-images.githubusercontent.com/19144982/149774885-a56b036d-089c-44e5-b97c-1efe6a647804.png)

我尝试去看preventCodes函数的作用，
猜测是当按下的按键能够匹配任意一个Shortcut后，将阻止默认行为，
但是我没有理解preventCodes中循环两层判断的matchCode是什么含义。

可能描述的不是很清楚，
想请教的问题是：
在您原本的设计上，有哪些情况下是需要阻止键盘默认行为呢？

**Comment by janryWang** [comment, marked as answer]
1. 快捷键序列解析的地方其实应该做个容错
2. codes的写法也应该是改成[[KeyCode.Shift]]，之前遗漏了

欢迎PR哈

**Reply by hou-pu** [reply]
2应该没事，parseCodes方法里兼容[KeyCode.Shift]写法，
1preventCodes方法的作用我没太搞清楚，所以不敢妄加改动
